iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -46,6 +46,49 @@ static void getMatmulConfig(SmallVectorImpl<TileWorkgroupSizePair> &tileSizes) {
   tileSizes.push_back(TileWorkgroupSizePair({{1, 128, 8}, {32, 1, 1}}));
 }
 
+/// Return the best combination of tile size and wg size when using tensorcore
+/// operations.
+static void getTensorCoreConfig(
+    SmallVectorImpl<TileWorkgroupSizePair> &tileSizes) {
+  tileSizes.push_back(TileWorkgroupSizePair({{64, 64, 16}, {64, 2, 1}}));
+}
+
+static std::string getTargetArch(FuncOp entryPoint) {
+  if (auto variantOp =
+          entryPoint->getParentOfType<IREE::HAL::ExecutableVariantOp>()) {
+    IREE::HAL::ExecutableTargetAttr targetAttr = variantOp.target();
+    if (auto config = targetAttr.getConfiguration()) {
+      if (auto attr = config.getAs<StringAttr>("target_arch")) {
+        return attr.getValue().str();
+      }
+    }
+  }
+  return "";
+}
+
+static bool supportsTensorCore(FuncOp entryPoint, linalg::LinalgOp op) {
+  // Limit tensor core pipeline to matmul as not all combinations of transpose
+  // are supported upstream.
+  // TODO(thomasraoux): Enable batchMatmul and generic contraction.
+  if (getTargetArch(entryPoint) != "sm_80" || !isa<linalg::MatmulOp>(op))
+    return false;
+  // Check that we support converting any fused operation. When using the
+  // tensorcore pipeline we need to be sure we can generate MMA ops otherwise
+  // the code will be highly inneficent.
+  bool fusedOpSupported = true;
+  entryPoint.walk([&fusedOpSupported](linalg::GenericOp linalgOp) {
+    for (Operation &fusedOp : linalgOp.getOps()) {
+      if (!isa<arith::AddFOp, arith::MulFOp, MaxFOp, MinFOp, linalg::YieldOp,
+               linalg::GenericOp, arith::DivFOp>(fusedOp)) {
```

Comment:
We cannot have `GenericOp` inside `GenericOp` anyway right? Isn't that verified by the op?